app/controllers/admin/blocked_ips_controller.rb

```diff
@@ -0,0 +1,46 @@
+#frozen_string_literal: true
+
+class Admin
+  class BlockedIpsController < ApplicationController
+    before_action :authenticate_user!
+    before_action :admin_required
+
+    layout "application"
+
+    def index
+      @q_blocked_by = params[:q_blocked_by].try( :strip )
+      @q_ip = params[:q_ip].try( :strip )
+
+      @blocked_ips = BlockedIp.joins( :user ).all
+      @blocked_ips = @blocked_ips.where( "users.login LIKE ?", "%#{@q_blocked_by}%" ) unless @q_blocked_by.blank?
+      @blocked_ips = @blocked_ips.where( "blocked_ips.ip LIKE ?", "%#{@q_ip}%" ) unless @q_ip.blank?
+      @blocked_ips = @blocked_ips.select( "blocked_ips.*, users.login as login" )
+      respond_to do | format |
+        format.html { render layout: "admin" }
+      end
+    end
+
+    def block
+      blocked_ip = BlockedIp.find_or_initialize_by( ip: params[:ip] )
+
+      unless blocked_ip.valid?
+        flash[:notice] = "Failed to block IP `#{ip}`: #{blocked_ip.errors.full_messages.to_sentence}"
```

Comment:
`ip` is not defined here, so when the IP is invalid and error is being thrown. `#{ip}` could be changed to `#{params[:ip]}`